crates/router/src/connector/payme/transformers.rs

```diff
@@ -1,109 +1,319 @@
 use serde::{Deserialize, Serialize};
 
 use crate::{
-    connector::utils::PaymentsAuthorizeRequestData,
+    connector::utils::{AddressDetailsData, CardData, PaymentsAuthorizeRequestData, RouterData},
     core::errors,
-    types::{self, api, storage::enums},
+    types::{self, api, storage::enums, MandateReference},
 };
 
-//TODO: Fill the struct with respective fields
-#[derive(Default, Debug, Serialize, Eq, PartialEq)]
-pub struct PaymePaymentsRequest {
-    amount: i64,
+#[derive(Debug, Serialize)]
+pub struct PayRequest {
+    buyer_name: Secret<String>,
+    buyer_email: pii::Email,
+    payme_sale_id: String,
+    #[serde(flatten)]
     card: PaymeCard,
 }
 
-#[derive(Default, Debug, Serialize, Eq, PartialEq)]
+#[derive(Debug, Serialize)]
+pub struct MandateRequest {
+    currency: enums::Currency,
+    sale_price: i64,
+    transaction_id: String,
+    product_name: String,
+    sale_return_url: String,
+    seller_payme_id: Secret<String>,
+    sale_callback_url: String,
+    buyer_key: String,
+}
+
+#[derive(Debug, Serialize)]
+#[serde(untagged)]
+pub enum PaymePaymentRequest {
+    MandateRequest(MandateRequest),
+    PayRequest(PayRequest),
+}
+
+#[derive(Debug, Serialize)]
 pub struct PaymeCard {
-    name: Secret<String>,
-    number: cards::CardNumber,
-    expiry_month: Secret<String>,
-    expiry_year: Secret<String>,
-    cvc: Secret<String>,
-    complete: bool,
+    credit_card_cvv: Secret<String>,
+    credit_card_exp: Secret<String>,
+    credit_card_number: cards::CardNumber,
+}
+
+#[derive(Debug, Serialize)]
+pub struct GenerateSaleRequest {
+    currency: enums::Currency,
+    sale_type: SaleType,
+    sale_price: i64,
+    transaction_id: String,
+    product_name: String,
+    sale_return_url: String,
+    seller_payme_id: Secret<String>,
+    sale_callback_url: String,
+    sale_payment_method: SalePyamentMethod,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct GenerateSaleResponse {
+    payme_sale_id: String,
+}
+
+impl<F, T>
+    TryFrom<types::ResponseRouterData<F, PaymePaySaleResponse, T, types::PaymentsResponseData>>
+    for types::RouterData<F, T, types::PaymentsResponseData>
+{
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(
+        item: types::ResponseRouterData<F, PaymePaySaleResponse, T, types::PaymentsResponseData>,
+    ) -> Result<Self, Self::Error> {
+        Ok(Self {
+            status: enums::AttemptStatus::from(item.response.sale_status),
+            response: Ok(types::PaymentsResponseData::TransactionResponse {
+                resource_id: types::ResponseId::ConnectorTransactionId(item.response.payme_sale_id),
+                redirection_data: None,
+                mandate_reference: item.response.buyer_key.map(|buyer_key| MandateReference {
+                    connector_mandate_id: Some(buyer_key.expose()),
+                    payment_method_id: None,
+                }),
+                connector_metadata: Some(
```

Comment:
This is closed sandbox and documentation, We are not sure whether this information would be used anywhere in further request during void. I feel like this could be some potential parameter so kept it in connector meta data